### PR TITLE
Fix `@children` in a PopupWindow

### LINKED ISSUE #5375: Placing @children in PopupWindow doesn't work
Platform: Linux, WASM
Language: Slint (Rust)

When using `@children` inside a `PopupWindow` the children aren't placed inside the popup window. This worked in 1.5 but doesn't work in 1.6 anymore.

Here is a slintpad with an code example: [example](https://slintpad.com/?snippet=component+Dropdown+inherits+Rectangle+%7B%0A%0A++++in-out+property+%3Cstring%3E+text%3B%0A%0A++++height%3A+2rem%3B%0A++++width%3A+10rem%3B%0A++++border-color%3A+black%3B%0A++++border-width%3A+1px%3B%0A%0A++++popup+%3A%3D+PopupWindow+%7B%0A++++++++x%3A+0px%3B%0A++++++++y%3A+2.1rem%3B%0A++++++++width%3A+10rem%3B%0A++++++++Rectangle+%7B%0A++++++++++++border-color%3A+black%3B%0A++++++++++++border-width%3A+1px%3B%0A++++++++++++VerticalLayout+%7B%0A++++++++++++++++padding%3A+0.5rem%3B%0A++++++++++++++++%40children%0A++++++++++++%7D%0A++++++++%7D%0A++++%7D%0A%0A++++Text+%7B%0A++++++++text%3A+text%3B%0A++++%7D%0A%0A++++t+%3A%3D+TouchArea+%7B%0A++++++++clicked+%3D%3E+%7B%0A++++++++++++popup.show%28%29%0A++++++++%7D%0A++++%7D%0A%7D%0A%0Aexport+component+Demo+%7B%0A++++width%3A+400px%3B%0A++++height%3A+300px%3B%0A%0A++++VerticalLayout+%7B%0A++++++++alignment%3A+start%3B%0A%0A++++++++Dropdown+%7B%0A++++++++++++text%3A+%22foo%22%3B%0A++++++++++++Text+%7B%0A++++++++++++++++text%3A+%22Option+1%22%3B%0A++++++++++++%7D%0A++++++++++++Text+%7B%0A++++++++++++++++text%3A+%22Another+Option%22%3B%0A++++++++++++%7D%0A++++++++%7D%0A++++%7D%0A%7D%0A&style=native)

And here is just the code example:
```
component Dropdown inherits Rectangle {

    in-out property <string> text;

    height: 2rem;
    width: 10rem;
    border-color: black;
    border-width: 1px;

    popup := PopupWindow {
        x: 0px;
        y: 2.1rem;
        width: 10rem;
        Rectangle {
            border-color: black;
            border-width: 1px;
            VerticalLayout {
                padding: 0.5rem;
                @children
            }
        }
    }

    Text {
        text: text;
    }

    t := TouchArea {
        clicked => {
            popup.show()
        }
    }
}

export component Demo {
    width: 400px;
    height: 300px;

    VerticalLayout {
        alignment: start;

        Dropdown {
            text: "foo";
            Text {
                text: "Option 1";
            }
            Text {
                text: "Another Option";
            }
        }
    }
}
```

The 1.6.0 release notes mention some work on placing `@children` in https://github.com/slint-ui/slint/issues/4935. Maybe this is related.

### PR DESCRIPTION
**Pull request by ogoffart**
Fixes #5375

### CODE REVIEW

File: internal/compiler/generator/rust.rs [hunk 1]
```diff
@@ -1044,6 +1044,7 @@ fn generate_sub_component(
             }
 
             pub fn user_init(self_rc: sp::VRcMapped<sp::ItemTreeVTable, Self>) {
+                #![allow(unused)]
```
Comment (ogoffart) [inline internal/compiler/generator/rust.rs:1047]: This is a drive by unrelated change i noticed by adding some debug code in some init=> in one of the tests

**Review by tronical — APPROVED** [review]
